python/lancedb/schema.py

```diff
@@ -0,0 +1,253 @@
+#  Copyright 2023 LanceDB Developers
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Schema related utilities."""
+
+import json
+from typing import Any, Dict, Type
+
+import pyarrow as pa
+
+
+def vector(dimension: int, value_type: pa.DataType = pa.float32()) -> pa.DataType:
```

Comment:
nit: should we wrap vector value type to `lance.FloatType`? That would make supporting bf16 easier I suppose? @wjones127 wdyt?

Reply:
We'll have a `lance.arrow.BFloat16Type` that subclasses `pa.ExtensionType`, which should subclass `pa.DataType`. So I think this is fine as is.